cmd/common/oci.go

```diff
@@ -330,6 +406,39 @@ func NewRunCommand(rootCmd *cobra.Command, runtime runtime.Runtime, hiddenColumn
 	return cmd
 }
 
+func runInstanceSpecsDetached(
+	ctx context.Context,
+	runtime runtime.Runtime,
+	specs []*gadgetmanifest.InstanceSpec,
+	runtimeParams *params.Params,
+	paramValueMap map[string]string,
+	runOptions ...gadgetcontext.Option,
+) error {
+	var merr error
+	for _, spec := range specs {
+		image := spec.Image
+
+		// Set some well-known params
+		runtimeParams.Set("id", spec.ID)
+		runtimeParams.Set("name", spec.Name)
+		runtimeParams.Set("tags", strings.Join(spec.Tags, ","))
+
+		maps.Copy(spec.ParamValues, paramValueMap)
+
+		gadgetCtx := gadgetcontext.New(
+			ctx,
+			image,
+			runOptions...,
+		)
```

Comment:
`gadgetCtx := gadgetcontext.New(ctx, image, runOptions...)`?

(I kind of feel bad commenting these things, but on the other hand I want to keep it as clean as possible)

Reply:
Don't feel bad about it.
I think I left it that way because it felt "more important" that way (the gadgetcontext being the central point of things), but I agree this should be shortened.